---
title: "GSE166261_CellBenderValidataion"
author: "Tawaun Lucas"
date: "12/7/2022"
output: 
  html_document:
    toc: true # table of content true
    toc_depth: 3  # upto three depths of headings (specified by #, ## and ###)
    toc_float: true
    number_sections: true  ## if you want number sections at each table header
    theme: united  # many options for theme, this one is my favorite.
    highlight: tango  # specifies the syntax highlighting style
---
# GSE166261 CellBender Validataion -- Shi P301S
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(out.width="100%")
library(Seurat)
library(SingleCellExperiment)
library(metaGP)
library(gpExtra)
library(dplyr)
library(gridExtra)
library(grid)
library(RColorBrewer)
library(ggplot2)
library(scDblFinder)
gac <- installedGeneAnnotationCollection("Neuroinflammation")
mouseHipp <- gac$`Mouse Hippocampus scRNASeq Markers`
ga.ctm <- geneSetList(mouseHipp) # "cell type markers"
ga.ctm <- lapply(ga.ctm, unlist) %>% lapply(unname)

```
## Set up funtions for analysis
<details>
<summary>Click for to see code</summary>
```{r Functions}
scoreCells <- function(data,seurat=FALSE,geneSetList=ga.ctm){
for (gene_set in names(geneSetList)) {
score_gene_set <- geneSetList[[gene_set]]
if(seurat==FALSE){
score_gene_set <- intersect(score_gene_set,rownames(data))
scores_by_cell <- colMeans(assay(data, 'logcounts')[score_gene_set,],na.rm = TRUE)
}
if(seurat==TRUE){
data1 <- as.SingleCellExperiment(data)
rownames(data1) <- features$ID
score_gene_set <- intersect(score_gene_set,rownames(data1))
scores_by_cell <- colMeans(assay(data1, 'logcounts')[score_gene_set,],na.rm = TRUE)
}
data[[gene_set]] <- scores_by_cell
}
return(data)
}

runSeurat <- function(obj){
obj <- NormalizeData(obj,verbose=FALSE)
obj <- FindVariableFeatures(obj)
obj <- ScaleData(object = obj, verbose=FALSE)
obj <- RunPCA(obj, npcs = 30,verbose=FALSE)
obj <- FindNeighbors(obj, reduction = "pca",dims=1:20,verbose=FALSE)
obj <- FindClusters(obj, resolution = 0.5,verbose=FALSE)
obj <- RunUMAP(obj,reduction= "pca",dims=1:20,verbose=FALSE)
print("Seurat Pipeline Finished!")
return(obj)
}
```
</details>



# WT data

```{r Load data into Seurat}
# load data from the original and filtered h5 file
raw <- Read10X_h5("/gstore/project/neurodegen_meta/data/cellbender/GSE166261/WT.h5", use.names = TRUE)
filtered <- Read10X_h5("/gstore/project/neurodegen_meta/data/cellbender/GSE166261/WT_filtered.h5", use.names = TRUE)

#Create Seurat objects
raw <- CreateSeuratObject(counts = raw)
filtered <- CreateSeuratObject(counts = filtered)

#lets only look at expression of the same barcoded cells
barcodes <- colnames(filtered)
raw <- raw[,colnames(raw) %in% barcodes]


feat <- as.data.frame(genomitory::getFeatures("GMTY17:GRCm38/GRCm38.IGIS4.0.genes.rds@REVISION-2"))
x <- match (rownames(raw),feat$symbol)
features <- feat[x,]
raw@assays$RNA@meta.features <- features
filtered@assays$RNA@meta.features <- features
```

## Adding Metadata
```{r}
raw$MitoPercent <- PercentageFeatureSet(raw, pattern = "^mt-")
raw$Condition <- "Wildtype"
raw$Genotype <- "Wildtype"
raw$StudyID <- "GSE166261"
raw$Sample <- "SRR13640892"
raw$Gender <- "Mixed"
raw$BrainRegion <- "Hippocampus"
raw$StudyName <- "Shi P301S"
raw$data.type <- "Single Nucleus"
raw$Age <- "9m"
raw$Disease <- "AD_Control"
raw$Model <- "Control"

filtered$MitoPercent <- PercentageFeatureSet(filtered, pattern = "^mt-")
filtered$Condition <- "Wildtype"
filtered$Genotype <- "Wildtype"
filtered$StudyID <- "GSE166261"
filtered$Sample <- "SRR13640892"
filtered$Gender <- "Mixed"
filtered$BrainRegion <- "Hippocampus"
filtered$StudyName <- "Shi P301S"
filtered$data.type <- "Single Nucleus"
filtered$Age <- "9m"
filtered$Disease <- "AD_Control"
filtered$Model <- "Control"

```


# P301S data
```{r Load P301S data into Seurat}
# load data from the original and filtered h5 file
P301S.raw <- Read10X_h5("/gstore/project/neurodegen_meta/data/cellbender/GSE166261/P301S.h5", use.names = TRUE)
P301S.filtered <- Read10X_h5("/gstore/project/neurodegen_meta/data/cellbender/GSE166261/P301S_filtered.h5", use.names = TRUE)

#Create Seurat objects
P301S.raw <- CreateSeuratObject(counts = P301S.raw)
P301S.filtered <- CreateSeuratObject(counts = P301S.filtered)

#lets only look at expression of the same barcoded cells
barcodes <- colnames(P301S.filtered)
P301S.raw <- P301S.raw[,colnames(P301S.raw) %in% barcodes]

P301S.raw@assays$RNA@meta.features <- features
P301S.filtered@assays$RNA@meta.features <- features
```

## Adding Metadata
```{r}
P301S.raw$MitoPercent <- PercentageFeatureSet(P301S.raw, pattern = "^mt-")
P301S.raw$Condition <- "P301S"
P301S.raw$Genotype <- "P301S"
P301S.raw$StudyID <- "GSE166261"
P301S.raw$Sample <- "SRR13640895"
P301S.raw$Gender <- "Mixed"
P301S.raw$BrainRegion <- "Hippocampus"
P301S.raw$StudyName <- "Shi P301S"
P301S.raw$data.type <- "Single Nucleus"
P301S.raw$Age <- "9m"
P301S.raw$Disease <- "AD"
P301S.raw$Model <- "P301S"

P301S.filtered$MitoPercent <- PercentageFeatureSet(P301S.filtered, pattern = "^mt-")
P301S.filtered$Condition <- "P301S"
P301S.filtered$Genotype <- "P301S"
P301S.filtered$StudyID <- "GSE166261"
P301S.filtered$Sample <- "SRR13640892"
P301S.filtered$Gender <- "Mixed"
P301S.filtered$BrainRegion <- "Hippocampus"
P301S.filtered$StudyName <- "Shi P301S"
P301S.filtered$data.type <- "Single Nucleus"
P301S.filtered$Age <- "9m"
P301S.filtered$Disease <- "AD"
P301S.filtered$Model <- "P301S"

```


# Merging the datasets for the two conditions

```{r warning=FALSE}
raw.data <- merge(x=raw,y=P301S.raw,add.cell.ids = c("WT", "P301S"))
raw.data@assays$RNA@meta.features <- features
rownames(raw.data@assays$RNA@meta.features)<- rownames(raw.data)
filtered.data <- merge(x=filtered,y=P301S.filtered,add.cell.ids = c("WT", "P301S"))
filtered.data@assays$RNA@meta.features <- features
rownames(filtered.data@assays$RNA@meta.features)<- rownames(filtered.data)

```

## Validataion Plots
```{r message=FALSE, warning=FALSE}
# Process raw datasets in Seurat
raw.data <- runSeurat(raw.data)
gc()
raw.data <- scoreCells(raw.data, seurat = TRUE)

filtered.data <- runSeurat(filtered.data)
filtered.data <- scoreCells(filtered.data, seurat = TRUE)

```

# Label Astrocytes
```{r Label Astrocytes}
library(SingleR)
library(BiocParallel)
#load reference dataset
ref <- DataSetDB::getDatasetAsSE("DS000002655") #NGS2722 labels
names(assays(ref)) <- c("counts","logcounts")

ref <- ref[,colSums(counts(ref)) > 0]
ref <- ref[,!is.na(ref$biologicalInterest)]
features <- rownames(ref)
symbols <- ref@rowRanges@elementMetadata@listData[["symbol"]]
ref <- as(ref,"SummarizedExperiment")
rownames(ref) <- symbols

filtered.dataSE <- as.SingleCellExperiment(filtered.data)
pred <- SingleR(test = filtered.dataSE, ref = ref,
                  labels = ref$biologicalInterest, de.method="wilcox", BPPARAM = MulticoreParam() )
filtered.data$FineLabels <- pred$labels
filtered.data$PrunedLabels <- pred$pruned.labels
```

## Plot Umaps
```{r}
p1 <- DimPlot(raw.data,label = TRUE, pt.size=1) + ggtitle("Raw Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold"),legend.position = "none")
p2 <- DimPlot(filtered.data,label=TRUE,pt.size=1) + ggtitle("Filtered Dataset")+
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold"),legend.position = "none")
grid.arrange( p1, p2,nrow=1,top = textGrob("Seurat Clusters",gp=gpar(fontsize=20,font=3)))
```

## Find Astrocytes
```{r message=FALSE, warning=FALSE}
grid.arrange( FeaturePlot(raw.data,features= "astrocyte",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Raw Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "Astrocyte") ,
 FeaturePlot(filtered.data,features= "astrocyte",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Filtered Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "Astrocyte"),
 nrow=1,top = textGrob("Astrocyte Scores",gp=gpar(fontsize=20,font=3)))

grid.arrange( DimPlot(filtered.data,group.by =  "FineLabels",reduction="umap", pt.size = 1, label = TRUE,repel = TRUE) + theme_void() + ggtitle("Cell Labels") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")),
 FeaturePlot(filtered.data,features= "astrocyte",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Filtered Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "Astrocyte"),
 nrow=1,top = textGrob("Astrocyte Scores",gp=gpar(fontsize=20,font=3)))


grid.arrange( FeaturePlot(raw.data,features= "microglia",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Raw Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "microglia") ,
 FeaturePlot(filtered.data,features= "microglia",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Filtered Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "microglia"),
 nrow=1,top = textGrob("Microglia Scores",gp=gpar(fontsize=20,font=3)))

grid.arrange( DimPlot(filtered.data,group.by =  "FineLabels",reduction="umap", pt.size = 1, label = TRUE,repel = TRUE) + theme_void() + ggtitle("Cell Labels") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")),
 FeaturePlot(filtered.data,features= "microglia",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Filtered Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "Astrocyte"),
 nrow=1,top = textGrob("Astrocyte Scores",gp=gpar(fontsize=20,font=3)))

```

### Expression Plots
```{r message=FALSE, warning=FALSE}
grid.arrange(
  VlnPlot(raw.data,features = "C1qa",group.by = "Model",log = TRUE) + ggtitle("Raw Dataset"),
  VlnPlot(filtered.data,features = "C1qa",group.by = "Model",log=TRUE) + ggtitle("Filtered Dataset"),
  nrow=1,top = textGrob("Expression of C1qa", gp=gpar(fontsize=20,font=3))
)


grid.arrange( FeaturePlot(raw.data,features= "C1qa",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Raw Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "C1qa Expression") ,
 FeaturePlot(filtered.data,features= "C1qa",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Filtered Dataset") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "C1qa Expression"),
 nrow=1,top = textGrob("C1qa Expression",gp=gpar(fontsize=20,font=3)))


```

# Doublet Detection
```{r Run Doublet Finder, message=FALSE}
filtered.sce <- as.SingleCellExperiment(filtered.data)
raw.sce <- as.SingleCellExperiment(raw.data)

filtered.sce <- scDblFinder(filtered.sce,nfeatures = 1500)
raw.sce <- scDblFinder(raw.sce,nfeatures = 1500)


```

# Plot Doublet results

```{r Transfer labels and Plot umap, echo=FALSE, message=FALSE}
filtered.data$scDblFinder.class <- filtered.sce$scDblFinder.class
filtered.data$scDblFinder.score <- filtered.sce$scDblFinder.score
filtered.data$scDblFinder.weighted <- filtered.sce$scDblFinder.weighted
filtered.data$scDblFinder.cxds_score <- filtered.sce$scDblFinder.cxds_score

raw.data$scDblFinder.class <- raw.sce$scDblFinder.class
raw.data$scDblFinder.score <- raw.sce$scDblFinder.score
raw.data$scDblFinder.weighted <- raw.sce$scDblFinder.weighted
raw.data$scDblFinder.cxds_score <- raw.sce$scDblFinder.cxds_score

grid.arrange(
  DimPlot(raw.data, group.by = "scDblFinder.class",pt.size = 1) + ggtitle("Raw Dataset"),
  DimPlot(filtered.data, group.by = "scDblFinder.class",pt.size = 1) + ggtitle("Filtered Dataset"),
  nrow=1,top = textGrob("Doublet Class",gp=gpar(fontsize=20,font=3)))

```

# Subset Astrocytes from filtered data
```{r Astrocytes, fig.show='hold', message=FALSE, warning=FALSE}

astros <- subset(filtered.data, subset = FineLabels=="astrocyte" )
astros <- runSeurat(astros)

grid.arrange(
  DimPlot(astros,label = TRUE, pt.size=1) + ggtitle("Seurat Clusters") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold"),legend.position = "none") ,
  FeaturePlot(astros,features= "astrocyte",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Astrocyte Score") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd"))+labs(colour = "Astrocyte"),
  nrow=1)

grid.arrange(
  FeaturePlot(astros,features = "Gfap",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Gfap") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd")),
  FeaturePlot(astros,features = "Aldh1l1",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Aldh1l1") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd")),
  nrow=1)

grid.arrange(
  FeaturePlot(astros,features = "Gja1",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Gja1") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd")),
  FeaturePlot(astros,features = "Aqp4",reduction="umap", pt.size = 1) +
  theme_void() + ggtitle("Aqp4") +
  theme(plot.title = element_text(hjust = 0.5,size=12,face = "bold")) +
  scale_colour_gradientn(colours = brewer.pal(n = 9, name = "YlOrRd")),
  nrow=1)

```

## Astrocyte numbers
```{r numbers}
print(paste0("There are a total of ", format(dim(astros)[2],big.mark=",",scientific=FALSE),
             " astrocytes.",
             " From which ", format(table(astros$Disease)["AD"],big.mark=",",scientific=FALSE),
             " are Diseased."))

print(paste0("There were a total of ", format(dim(filtered.data)[2],big.mark=",",scientific=FALSE),
             " cells."))

print(paste0("Astrocytes account for ", format( round((dim(astros)[2]/dim(filtered.data)[2])*100,2), big.mark=",",scientific=FALSE),
             "% of cells."))


```



# Save Dataset for further processing

```{r Saving dataset}
saveRDS(astros,"/gstore/project/neurodegen_meta/data/cellbender/GSE166261/Astrocytes.RDS")
saveRDS(filtered.data,"/gstore/project/neurodegen_meta/data/cellbender/GSE166261/filteredData.RDS")
saveRDS(raw.data,"/gstore/project/neurodegen_meta/data/cellbender/GSE166261/rawData.RDS")
```

















