---
title: "Local_analyses"
output: html_document
date: "2024-06-18"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Decide which clusters to remove
```{r}
clusters <- readRDS("/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/integration_with_cellbender_counts/Clusters_res3.not_cleaned_space.rds")

scores <- readRDS("/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/integration_with_cellbender_counts/Cell_types_scores_for_space_cleaning.rds") ### getting updated with all cell types

```


```{r}

cluster_info <- data.frame(cluster = clusters$membership)
names <- names(scores)

get_results <- function(cluster_info, name) {
    formula=as.formula("celltype~ cluster")
    tmp <- data.frame(cluster = cluster_info$cluster, celltype = scores[[name]])
    res <- aggregate(formula, tmp, median )
    p <- ggplot(data=res, aes_string(x="cluster", y="celltype")) + geom_bar(stat="identity") + theme_classic() + ggtitle(name) + theme(plot.title = element_text(hjust = 0.5))

 return(p)

}

all_plots <- list()

for ( i in 1:length(names)) {
  p <- get_results(cluster_info, names[i])
  all_plots[[i]] <- p
}

library(gridExtra)

png("/gne/web/dev/apache/htdocs/people/novikovg/Astrocytes_meta/Harmony_AD_MS_PD_integration_2024//Marker_analysis_clustering_plots.png",width = 3000, height = 1200)
do.call("grid.arrange", c(all_plots, ncol=5))
dev.off()

```

UMAPs highlihgting each cluster are generated with this: get_UMAPs_with_clusters_highlighted.R

clusters 2, 28, 34 - low astrocyte score
cluster 6 - microglia/PVM
cluster 35 - neuron
cluster 1 - oligos
cluster 2 - ependymal


```{r}
out <- readRDS("/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/integration_with_cellbender_counts/AD_MS_PD_astrocytes.cellbender_counts.20cell_filter.with_logcounts.v3.liddelow_cellranger_rest_cellbender.rds")

out_clean <- out[,! out$clusters %in% c(1,2, 6, 28, 34, 35)]

reducedDims(out_clean) <- list()

assays(out_clean) <- assays(out_clean)[1]

donors_to_remove <- names(table(out_clean$donor) < 20)[as.logical(table(out_clean$donor) < 20)]

out_clean_final <- out_clean[,! out_clean$donor %in% donors_to_remove ]


saveRDS(out_clean_final, "/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/integration_with_cellbender_counts/AD_MS_PD_astrocytes.CLEAN_from_non_astrocytes.cellbender_counts.20cell_filter.no_logcounts.liddelow_cellranger_rest_cellbender.rds")


```

## figure out the # of unique donors
```{r}

## DONE "SEA-AD"  "absinta"  "bryois"   "cain"     "gerrits"  "jakel"  "kamath"   "liddelow" "morabito"  "schirmer"  "smajic" "smith"    "wang" 

coldata <- as.data.frame(colData(out))

## SEA-AD
n_sea_ad <- coldata %>% dplyr::filter(studybatch == "SEA-AD") %>% dplyr::select(donor) %>% unique %>% nrow

## Absinta
n_absinta <- coldata %>% dplyr::filter(studybatch == "absinta") %>%  dplyr::select(sex, diagnosis_harmonized, age) %>% drop_na() %>% mutate(donorIDunique = paste(sex, diagnosis_harmonized, age, sep="_")) %>% dplyr::select(donorIDunique) %>% unique %>% nrow
coldata <- coldata %>% dplyr::mutate(donorIDunique = ifelse(studybatch == "absinta", paste("absinta", sex, diagnosis_harmonized, age, sep="_"), donor))

## Bryois
n_bryois <- unlist(lapply(coldata[coldata$studybatch=="bryois" ,]$donor %>% unique, function(x) strsplit(x, "-")[[1]][1])) %>% unique %>% length
coldata[coldata$studybatch == "bryois",]$donorIDunique <- unlist(lapply(coldata[coldata$studybatch=="bryois" ,]$donor, function(x) strsplit(x, "-")[[1]][1]))

## Cain
n_cain <- coldata %>% dplyr::filter(studybatch == "cain") %>% dplyr::select(donor) %>% unique %>% nrow

## Gerrits
library(stringr)
coldata[coldata$studybatch == "gerrits",]$donorIDunique <-  paste("gerrits", str_replace(coldata[coldata$studybatch=="gerrits",]$donor,"OTC|OC",""), sep="_")
n_gerrits <- coldata %>% dplyr::filter(studybatch == "gerrits") %>% dplyr::select(donorIDunique) %>% unique %>% nrow

### Jakel
coldata[coldata$studybatch == "jakel",]$donorIDunique <- paste("jakel", unlist(lapply(coldata[coldata$studybatch=="jakel" ,]$donor, function(x) strsplit(x, "_")[[1]][2])), sep="_")
n_jakel <- coldata %>% dplyr::filter(studybatch == "jakel") %>% dplyr::select(donorIDunique) %>% unique %>% nrow

## Kamath

n_kamath <- coldata %>% dplyr::filter(studybatch == "kamath") %>% dplyr::select(donor) %>% unique %>% nrow

## liddelow

n_liddelow <- coldata %>% dplyr::filter(studybatch == "liddelow") %>% dplyr::select(donor) %>% unique %>% nrow

## morabito 

n_morabito <- coldata %>% dplyr::filter(studybatch == "morabito") %>% dplyr::select(donor) %>% unique %>% nrow

## schirmer
n_schirmer <- coldata %>% dplyr::filter(studybatch == "schirmer") %>% dplyr::select(donor) %>% unique %>% nrow

## smajic
n_smajic <- coldata %>% dplyr::filter(studybatch == "smajic") %>% dplyr::select(donor) %>% unique %>% nrow

## smith
n_smith <- coldata %>% dplyr::filter(studybatch == "smith") %>% dplyr::select(donor) %>% unique %>% nrow

# wang

n_wang <- coldata %>% dplyr::filter(studybatch == "wang") %>% dplyr::select(donor) %>% unique %>% nrow

total_unique_donors <- n_sea_ad + n_absinta + n_bryois + n_cain + n_gerrits + n_jakel + n_kamath + n_liddelow + n_morabito + n_schirmer + n_smajic + n_smith + n_wang 


```


```{r}

get_umap_plot <- function(harmony_coords_final,dataset) {
  ### can be used for dataset specific plotting, but also for clusters
  p <- ggplot(data=harmony_coords_final, aes(x=UMAP1, y=UMAP2)) +
    geom_point(color="grey", size=0.1,alpha = 1) +
    geom_point(data = harmony_coords_final %>% dplyr::filter((!!sym(dataset))==1), color="darkgreen", size=0.5, alpha=0.5)+
    theme_classic() + ggtitle(dataset) + theme(plot.title = element_text(hjust = 0.5)) +
    theme(plot.title = element_text(size = 20, face = "italic"))
  return(p)
}

get_harmony_coords <- function(se, harmony_umap_dim) {
  message(paste(names(reducedDims(se))[harmony_umap_dim]))
  harmony_coords_final <- reducedDims(se)[[harmony_umap_dim]]
  colnames(harmony_coords_final) <- c("UMAP1","UMAP2")
  harmony_coords_final <- as.data.frame(cbind(harmony_coords_final,colData(se)))

  results <- fastDummies::dummy_cols(se$studybatch)
  colnames(results) <- unlist(lapply(colnames(results),function(x) strsplit(x,".data_")[[1]][2]))
  results <- results[,-1]
  harmony_coords_final <- cbind(harmony_coords_final, results)

  return(harmony_coords_final)

}

harmony_coords <- get_harmony_coords(out,3)


list_of_plots <- list()

studies <- unique(out$studybatch)

library(ggplot2)

for (i in 1:length(studies)) {
  p <- get_umap_plot(harmony_coords,studies[i])
  list_of_plots[[i]] <- p
}

library(gridExtra)
png("/gne/web/dev/apache/htdocs/people/novikovg/Astrocytes_meta/Harmony_AD_MS_PD_integration_2024/Studies_UMAP.theta_1.5.png",width = 4500, height = 2000, res = 250)
do.call("grid.arrange", c(list_of_plots, ncol=8))
dev.off()


```
Generate a pseudobulk file merged by donorIDunique (regions/sections merged together)
```{r}

agg <- readRDS("/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/DEG/aggregated_cells.AD_MS_PD.062424.rds")

rowdata_agg_cluster <- data.frame(ID = rownames(agg$sums))
rowdata_agg_cluster_final <- left_join(rowdata_agg_cluster, as.data.frame(rowData(out)) %>% dplyr::select(ID, symbol), by = "ID")

coldata_agg <- as.data.frame(agg$combinations)

coldata_agg_final <-  left_join(coldata_agg, coldata %>% dplyr::select(age, diagnosis_harmonized, sex, studybatch,donor, batch, donorIDunique) %>% unique, by = "donor")

se_agg <- SingleCellExperiment(list(counts=agg$sums),
    colData=coldata_agg_final,
    rowData=rowdata_agg_cluster_final)

### calculate another aggregated object, but based on donorIDunique
x <- initializeSparseMatrix(assay(se_agg, 1), num.threads = 14)
agg_donorunique <- aggregateAcrossCells.chan(x, list(donorIDunique=se_agg$donorIDunique), num.threads=14)

rowdata_agg_donorunique_cluster <- data.frame(ID = rownames(agg_donorunique$sums))
rowdata_agg_donorunique_cluster_final <- left_join(rowdata_agg_donorunique_cluster, as.data.frame(rowData(out)) %>% dplyr::select(ID, symbol), by = "ID")

coldata_donorunique_agg <- as.data.frame(agg_donorunique$combinations)

coldata_donorunique_agg_final <-  left_join(coldata_donorunique_agg, coldata %>% dplyr::select(age, diagnosis_harmonized, sex, studybatch, donorIDunique) %>% unique, by = "donorIDunique") ## add batch also 

se_donorunique_agg <- SingleCellExperiment(list(counts=agg_donorunique$sums),
    colData=coldata_donorunique_agg_final,
    rowData=rowdata_agg_donorunique_cluster_final)

y <- DGEList(assay(se_donorunique_agg, 1))
y$genes <- data.frame(origin=seq_len(nrow(se_donorunique_agg)))
y <- calcNormFactors(y)
se_donorunique_agg$lib.size <- y$samples$lib.size
se_donorunique_agg$norm.factors <- y$samples$norm.factors
assay(se_donorunique_agg, "logCPM") <- normalizedCPM(assay(se_donorunique_agg, 1), lib.size=se_donorunique_agg$lib.size,
    norm.factors=se_donorunique_agg$norm.factors, offset=y$offset)

saveRDS(se_donorunique_agg, "/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/DEG/aggregated_cells.AD_MS_PD.062624.DonorUniqueID_regions_merged.SE.rds")

```

AD differential expression

```{r}

res_ad <- list()

### Let's run DE for all datasets except the ones that have internal batches - marabito & cain

studies <- unique(se_donorunique_agg[,se_donorunique_agg$diagnosis_harmonized == "AD"]$studybatch)

for (i in 1:length(studies)) {
  study <- studies[i]
   out_tmp <- gp.sa.diff::runVoom(se_donorunique_agg, groups="diagnosis_harmonized", comparisons=list(c("AD", "Control")), subset.factor="studybatch",
    subset.levels=c(study), commit="never")

   out_final <- as.data.frame(out_tmp[[1]]) %>% mutate(symbol = rowData(se_donorunique_agg)$symbol)

   res_ad[[i]] <- out_final
}

## ADD BATCHES LATER
#out_cain <- as.data.frame(runVoom(se_donorunique_agg[,se_donorunique_agg$studybatch == "cain"], groups="diagnosis_harmonized", comparisons=list(c("AD", "Control")), block = "batch", commit="never")[[1]]) %>% mutate(symbol = rowData(se_pseud)$symbol)
#out_morabito <- as.data.frame(runVoom(se_donorunique_agg[,se_donorunique_agg$studybatch == "morabito"], groups="diagnosis_harmonized", comparisons=list(c("AD", "Control")), block = "batch", commit="never")[[1]]) %>% mutate(symbol = rowData(se_pseud)$symbol)
#n <- length(studies)
#res[[n+1]] <- out_cain
#res[[n+2]] <- out_morabito
#names(res) = c(studies,"cain","morabito")
#n_studies <- length(res)

names(res_ad) <- studies

res <- res_ad

feat <- as.data.frame(genomitory::getFeatures("GMTY17:GRCh38/GRCh38.IGIS4.0.genes.rds@REVISION-3"))

fc <- matrix(0, ncol=length(res), nrow=dim(feat)[1])
rownames(fc) <- feat$ID

pval <- matrix(NA, ncol=length(res), nrow=dim(feat)[1])
rownames(pval) <- feat$ID

mexp <- matrix(NA, ncol=length(res), nrow=dim(feat)[1])
rownames(mexp) <- feat$ID

sefc <- matrix(NA, ncol=length(res), nrow=dim(feat)[1])
rownames(sefc) <- feat$ID

for(i in 1:length(res)){
        m<-match(rownames(fc), rownames(res[[i]]))
        f.a=!is.na(m)
        f.b=m[f.a]
        fc[f.a,i] <- res[[i]][f.b,"LogFC"]
        pval[f.a,i] <- res[[i]][f.b,"PValue"]
        mexp[f.a,i] <- res[[i]][f.b,"AveExpr"]
        sefc[f.a,i] <- res[[i]][f.b,"LogFC"]/res[[i]][f.b,"t"]
}

x<-strsplit(names(res),".", fixed=T)
x2 <- sapply(x, function(x) purrr::pluck(x,1))
colnames(pval) = x2
colnames(fc) = x2
colnames(mexp) = x2
colnames(sefc) = x2


deg_master <- list(pval=pval, fc=fc, se_fc=sefc, mean_exp=mexp)


y <- strsplit(colnames(deg_master$pval), "_", fixed = T)

y2 <- lapply(y, function(x)
  purrr::pluck(x, 2))

res <- vector("list", length = 1)


f <- colnames(deg_master$pval)
pval <- deg_master$pval[, f]
fc <- deg_master$fc[, f]
mexp <- deg_master$mean_exp[, f]
sefc <- cbind(fc,deg_master$se_fc[,f])

pval_up <- pval * 0
pval_down <- pval * 0

for (i in 1:dim(fc)[2]) {
    f1 <- which(fc[, i] > 0)
    f2 <- which(fc[, i] < 0)
    pval_up[f1, i] <- pval[f1, i] * 0.5
    pval_up[f2, i] <- 1 - (pval[f2, i] * 0.5)
    pval_down[f1, i] <- 1 - (pval[f1, i] * 0.5)
    pval_down[f2, i] <- pval[f2, i] * 0.5

  }

library(metafor)
library(metap)
meta_up <- suppressWarnings(apply(pval_up, 1, function(y) sumlog(y[!is.na(y)])$p))
meta_down <- suppressWarnings(apply(pval_down, 1, function(y) sumlog(y[!is.na(y)])$p))
mean_fc <- rowMeans(fc, na.rm = T)
mean_exp <- rowMeans(mexp, na.rm = T)

n_studies <- length(studies)

f_se_fc <- apply(sefc[,1:n_studies],1,function(x) sum(!is.na(x))>=2)
f_se_se <- apply(sefc[,(n_studies+1):(n_studies*2)],1,function(x) sum(!is.na(x))>2)
f_se <- f_se_fc & f_se_se

dsl_res <- apply(sefc[f_se,], 1, function(row) rma(yi=as.vector(row[1:n_studies]),vi=as.vector(row[(n_studies+1):(n_studies*2)]),method="DL"))

betas <- sapply(dsl_res,function(x) purrr::pluck(x,"beta"))
ses <- sapply(dsl_res,function(x) purrr::pluck(x,"se"))
dl_ps <- sapply(dsl_res,function(x) purrr::pluck(x,"pval"))
dl_ps <- sapply(dsl_res,function(x) purrr::pluck(x,"pval"))

dl_sub <- data.frame(ID=rownames(sefc)[f_se],beta=betas,se=ses,pval=dl_ps)

dl_all <- data.frame(ID=rownames(sefc))
dl_all <- left_join(dl_all,dl_sub[,c("ID","beta","se","pval")],by=c("ID"="ID"))

  temp <-
    data.frame(
      ID = names(meta_up),
      AveExpr = as.numeric(mean_exp),
      PValue = dl_all$pval,
      FDR = p.adjust(dl_all$pval,method="BH"),
      LogFC = as.numeric(mean_fc),
      dl_mu = dl_all$beta,
      dl_s = dl_all$se,
      metap_up = meta_up,
      metap_down = meta_down,
      adj_metap_up = p.adjust(meta_up, method = "BH"),
      adj_metap_down = p.adjust(meta_down, method = "BH"),
      n_tested = apply(pval_up, 1, function(y)
        sum(!is.na(y))),
      n_up = apply(fc, 1, function(y)
        sum(y > 0, na.rm = T)),
      n_down = apply(fc, 1, function(y)
        sum(y < 0, na.rm = T))
    )

  temp <- left_join(temp, feat, by = c("ID" = "ID"))
  ord <- order(temp$metap_up)

  temp <- temp[ord, ]
  x <-
    c(
      "ID",
      "symbol",
      "AveExpr",
      "PValue",
      "FDR",
      "LogFC",
      "dl_mu",
      "dl_s",
      "metap_up",
      "metap_down",
      "adj_metap_up",
      "adj_metap_down",
      "n_tested",
      "n_up",
      "n_down",
      "symbol"
    )

res_dsl <- temp %>% dplyr::select(x)

saveRDS(res_dsl, "/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/DEG/res_dsl.AD.062624.rds")



```


```{r}
res_dsl <- res_dsl %>% dplyr::mutate(adj_metap_overall = ifelse(dl_mu>0,adj_metap_up,
                                                                ifelse(dl_mu<0, adj_metap_down, NA)))
res_dsl <- res_dsl %>% dplyr::mutate(diffexpressed = ifelse(dl_mu>=0.5 & adj_metap_up<=0.05 & n_up>=3, "up",
                                ifelse(dl_mu<= -0.5 & adj_metap_down<=0.05 & n_down>=3, "down", "unchanged")))

get_volcano_plot <- function(res) {
  res <- res %>% drop_na()
  res_dsl <- res_dsl %>% dplyr::filter(AveExpr>0)
  p_volcano <- ggplot(data=res, aes(x=dl_mu, y=-log10(adj_metap_overall), col=diffexpressed, label=symbol)) +
        geom_point() +
        theme_classic() + theme(axis.title=element_text(size=12), axis.text=element_text(size=12),legend.text=element_text(size=10)) +
        geom_text_repel(data = res[res$diffexpressed %in% c("up","down"),],size=3, aes(x=dl_mu,y=-log10(adj_metap_overall)),show.legend = FALSE,box.padding = 0.5, max.overlaps = 40) +
        scale_color_manual(values=c("blue", "black", "red")) + xlab("MetaLogFoldChange") + ylab(bquote(log[10](MetaPValue)))

  return(p_volcano)
}

png("/gne/web/dev/apache/htdocs/people/novikovg/Astrocytes_meta/Harmony_AD_MS_PD_integration_2024/AD_DEG_volcano_plot.png", res = 300, width = 2000, height = 1500)
get_volcano_plot(res_dsl)
dev.off()

# get_gene_plot <- function(processed.se, gene) {
# 
#   ens <- rownames(rowData(processed.se)[rowData(processed.se)$symbol == gene,])
# 
#   df <- data.frame(expr = as.numeric(assay(processed.se, "logCPM")[ens,]),
#                    Group = processed.se$TREATMENT_NAME)
# 
#  plot <-  ggplot(df, aes(x=Group, y=expr,fill=Group)) + theme_classic() +geom_boxplot(outlier.shape=NA) + geom_jitter(width = 0.1)+ theme(axis.text.x = element_text(angle = 65, vjust = 0.5)) + scale_fill_manual(values = cols, name = "Group") + xlab("")  + ggtitle(gene) + theme(plot.title = element_text(hjust = 0.5))
# 
#  return(plot)
# 
# }


```

4-way between old and new meta 

```{r}

merged_res_dsl <- dplyr::inner_join(res_dsl_old, res_dsl, by = "ID", suffix = c(".old", ".new"))

merged_res_dsl <- merged_res_dsl %>% dplyr::mutate(sig.overall = ifelse(diffexpressed.old %in% c("up","down") & ! diffexpressed.new %in% c("up","down"), "Only in old meta", 
ifelse(! diffexpressed.old %in% c("up","down") & diffexpressed.new %in% c("up","down"), "Only in new meta", 
ifelse( diffexpressed.old %in% c("up","down") & diffexpressed.new %in% c("up","down"), "Changed in both", "Neither")))) 

merged_res_dsl <- merged_res_dsl %>% drop_na()

plot_4way_res_dsls <- ggplot(merged_res_dsl, aes(x=dl_mu.old, y = dl_mu.new, color=sig.overall)) +
  geom_point() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(), axis.line = element_line(colour = "black")) +

  scale_color_manual(values=c("forestgreen", "grey","darkred", "blue")) +
  geom_hline(yintercept =0) +
  geom_vline(xintercept=0) +
  geom_text_repel(data = merged_res_dsl[merged_res_dsl$sig.overall != "Neither" ,],size=3, aes(x=dl_mu.old,y=dl_mu.new,label=symbol.old),
  show.legend = FALSE,box.padding = 0.5, max.overlaps = 20) +
  xlab("dl_mu old meta") + ylab("dl_mu new meta") + theme(plot.title = element_text(hjust = 0.5)) + guides(fill=guide_legend(title=""))


png("/gne/web/dev/apache/htdocs/people/novikovg/Astrocytes_meta/Harmony_AD_MS_PD_integration_2024/4way_old_new_AD_DEGs.png", res = 300, width = 2000, height = 1500)
plot_4way_res_dsls
dev.off()

library(clusterProfiler)
library(enrichplot)

ad_up_candidates <- res_dsl %>% dplyr::filter(diffexpressed == "up")
ad_down_candidates <- res_dsl %>% dplyr::filter(diffexpressed == "down")

res_dsl_no_na <- res_dsl %>% drop_na()

go_enrich_ad_up_ALL <- enrichGO(gene = unique(ad_up_candidates$symbol),
                      universe = unique(res_dsl_no_na$symbol),
                      OrgDb = "org.Hs.eg.db",
                      keyType = 'SYMBOL',
                      readable = T,
                      ont = "ALL",
                      pvalueCutoff = 0.05,
                      qvalueCutoff = 0.10)
go_enrich_ad_down_ALL <- enrichGO(gene = unique(ad_down_candidates$symbol),
                      universe = unique(res_dsl_no_na$symbol),
                      OrgDb = "org.Hs.eg.db",
                      keyType = 'SYMBOL',
                      readable = T,
                      ont = "ALL",
                      pvalueCutoff = 0.05,
                      qvalueCutoff = 0.10)

go_enrich_ad_up_MF <- enrichGO(gene = unique(ad_up_candidates$symbol),
                      universe = unique(res_dsl_no_na$symbol),
                      OrgDb = "org.Hs.eg.db",
                      keyType = 'SYMBOL',
                      readable = T,
                      ont = "MF",
                      pvalueCutoff = 0.05,
                      qvalueCutoff = 0.10)



```

Look at candidates

```{r}


colors <- c(adjustcolor('#31a354', alpha.f = 0.9), '#8856a7', adjustcolor('#fa9fb5', 0.9),  '#2b8cbe', adjustcolor('#fec44f', alpha.f = 0.9), adjustcolor('#c51b8a', 0.9),  adjustcolor('#d95f0e', alpha.f = 0.9), adjustcolor('turquoise1', alpha.f = 0.9))

get_plot_for_gene <- function(se_donorunique_agg, genename, colors) {

  gene = rowData(se_donorunique_agg)[rowData(se_donorunique_agg)$symbol == genename,]$ID
  
  df_mic <- data.frame(expr = t(assay(se_donorunique_agg[gene,],2)), diagnosis = se_donorunique_agg$diagnosis_harmonized, studybatch = se_donorunique_agg$studybatch) %>% dplyr::filter(diagnosis %in% c("AD", "Control"))
  df_mic$diagnosis <- factor(df_mic$diagnosis, levels = c("Control","AD"))

  p <- ggplot(df_mic, aes(diagnosis,!!sym(gene), fill=diagnosis)) + scale_fill_manual(values = c("gray43", "dodgerblue3")) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(aes(colour=studybatch), width = 0.1)+
    theme(legend.position = "none") + theme_classic() + theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5)) + ylab("logCPM") + scale_color_manual(values = colors) + facet_wrap(~ studybatch)

  return(p)
}



```

check out markers for harmony integration with studybatch as group vars

```{r}
rowdata_ast <- readRDS("/gstore/data/omni/neuroscience/AD/Astrocytes_AD_MS_PD_meta/AD_MS_PD_cellbender_integration/harmony_integration/integration_with_cellbender_counts/AD_MS_PD_astrocytes.CLEAN_from_non_astrocytes.cellbender_counts.20cell_filter.ROWDATA.rds")

m.out.0.5 <- readRDS("/gstore/scratch/u/novikovg/Astrocytes_meta/AD_MS_PD_cellbender_harmony_2024/harmony_integration/clustering/clustering_of_clean_space_with_studybatch_as_groupvars/m.out.0.5.rds")


m.out.0.5.with_symbols <- list()
for (i in (1:length(m.out.0.5$statistics))) {
  
  tmp <- m.out.0.5$statistics[[i]]
  tmp$ID <- rownames(tmp)
  rowdata_ast <- as.data.frame(rowdata_ast) %>% dplyr::select(ID, symbol)
  tmp <- left_join(tmp, rowdata_ast, by  = "ID")
  
  m.out.0.5.with_symbols[[i]] <- as.data.frame(tmp)
  
}

```


